clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -266,8 +270,7 @@
  *                 for (ConsumerRecord&lt;String, String&gt; record : partitionRecords) {
  *                     System.out.println(record.offset() + &quot;: &quot; + record.value());
  *                 }
- *                 long lastOffset = partitionRecords.get(partitionRecords.size() - 1).offset();
- *                 consumer.commitSync(Collections.singletonMap(partition, new OffsetAndMetadata(lastOffset + 1)));
+ *                 consumer.commitSync(Collections.singletonMap(partition, partitionRecords.nextOffsets().get(partition));
```

Comment:
typo: `partitionRecords` -> `records`

Reply:
and missing parenthesis, I guess we should end with :
```suggestion
 *                 consumer.commitSync(Collections.singletonMap(partition, records.nextOffsets().get(partition)));
```

Reply:
`partitionRecords` is actually correct; `nextRecords` was added to `ConsumerRecord[s]` (not `ConsumerRecord`) -- that's also why we need to call `get(partition)` -- `ConsumerRecords` is already data fro a single partition.

Reply:
@mjsax Could you please take a look at following attachment? the type of `partitionRecords` is `List<ConsumerRecord<String, String>>` rather than `ConsumerRecords`

![Screenshot From 2025-01-03 10-42-00](https://github.com/user-attachments/assets/a3c93e0f-09c1-4dfd-aae7-e616b2dce01d)

btw, it miss a `)` also.

Reply:
Thanks for sanity checking... I did mix up the variable names.